### DISCUSSION TITLE
Multi-language management of dashboards

### DISCUSSION THREAD

**Discussion by renattissimo** [perses/perses#2112] [Ideas]
One use case is to create dashboards that can be used for several clients in different regions with different languages. I don't know if something is already along this line, but it would be interesting to have the labels in various languages and use some switch like /es or some other mechanism, according to the client.

**Comment by Nexucis** [comment]
Hello,

That sounds difficult to me to achieve that as we don't own the data that are coming from various datasources like Prometheus. So I don't really know how we could translate the data at the runtime, specially since we don't know what in which language labels, metrics, traces/spans, logs will be written

**Reply by renattissimo** [reply]
I was thinking of having a JSON with the custom translations for each language. There, I could have the descriptions' texts and the charts' names. In the first instance, this is a solution that doesn't impact the core so much.

**Reply by Nexucis** [reply]
yeah that's usually how it works when supporting multi language for static text. But that doesn't solve how we can know in which language the data are written.
That's something we can do for the few texts/sentences put in Perses itself, but to translate data coming from external datasource, how can we know the language ? 

> In the first instance, this is a solution that doesn't impact the core so much.

Well I assume the UI needs to use it to perform the translation, so it impacts the core for sure at least from my point of view

**Reply by renattissimo** [reply]
The use case I propose is less focused on the language in which the data is written, assuming that most of it is numeric or that we don't want to touch it for now. It is more focused on the language of our client, where the data may be the same, but the names of the tables, descriptions, and titles may change the language. It's more on the front-end side. From a development point of view, the idea is not to have N copies of the same dashboard where N = the number of languages you want to translate into. Having one dashboard with one set of possible languages would make it easier to maintain them.

**Reply by Nexucis** [reply]
aaaah ok, thank you for the clarifications. It makes more sense to me know.

I am not super familiar with how internationalisation works, but I think if someone want to work on it, we can definitely consider it.
It will probably require a design first on how it is supposed to work before proposing a code.

**Reply by UniqueEgret** [reply]
We also have a need for internationalization, may I ask, is there a next version planned for internationalization?

**Reply by Nexucis** [reply]
So it's not planned as we are not enough to tackle everything. But like I said in a previous message, if someone would like to work on it, this person can start by proposing a design. @andreasgerstmayr started another thread here: https://github.com/perses/perses/discussions/2287 slightly related to this topic. So perhaps something will start around this topic. 

Let's see.

If not and if there is no one from the community that would like to do it, then, it will have to wait.

**Reply by yosiasz** [reply]
I would be more than happy to contribute 😁

**Reply by Nexucis** [reply]
feel free to draft a proposition @yosiasz :). Probably the first thing would be to specify the scope of the internationalisation. Does it include the plugin for example ?

**Reply by yosiasz** [reply]
Light but gets the wheel spinning @Nexucis 

```
Scope 
	Core UI first
	Long term: plugins
	
Translation source 
	service: api/translate/fr 
           Pro: no local language files
           Con: dependency
	local files: 
                Pro: fast,reliable
                Con: all language files, loaded
		human generated?
		auto generated? First pass an automated process goes through the whole app and generates literals?
		
		literals.draft.yaml
			- Perses_Common_LblActive: "Active <!-- label, filter option, status -->"
			- Perses_Common_HdgFoo: "Foo <!-- heading -->"			
		literals.yaml
		
		Generates: i18n/en.json, fr.json, es.json
			{"Perses_Common_LblActive": "Active", "Perses_Common_HdgActive" : "Foo" }
	Possible Implementation
            <Stack direction="row" alignItems="center" gap={1}>
              <ViewDashboardIcon />
              <h1><Trans i18nKey="Perses_lblImportantDashboards"/></h1>
            </Stack>

Date/time formats, number and currency formats, and even text direction 


Library
	i18next - https://react.i18next.com/latest/usetranslation-hook
	React Intl (FormatJS)
POC:
	implement in one corner of the app	
```

**Reply by yosiasz** [reply]
https://github.com/perses/perses/pull/3103

POC

<img width="793" height="472" alt="image" src="https://github.com/user-attachments/assets/b3f6842e-2d04-4e4c-8f3c-07dd6724c1f7" />

**Reply by yosiasz** [reply]
this is ready to roll if someone can take a peek?

https://github.com/perses/perses/pull/3103
